src/python/pants/engine/target.py

```diff
@@ -467,9 +467,9 @@ def field_types(self) -> Tuple[Type[Field], ...]:
     @final
     @memoized_classproperty
     def _plugin_field_cls(cls) -> type:
-        # NB: We ensure that each Target subtype has its own `PluginField` class so that
-        # registering a plugin field doesn't leak across target types.
-
+        # Use the `PluginField` of the first `Target`-subclass ancestor as a base to ours, so that
```

Comment:
Also, subject for debate, why not _all_ target bases?

Reply:
Was going off of the assumption that you'd only inherit from a single Target base class.. what would it mean if you did
```python
class MyTarget (SomeTarget, AnotherTarget):
   ...
```

but sure enough, if you'd venture out and try that, you'd get potentially surprising results (but I'd guess this wouldn't be the only one, though)

but now thanks to #16657 I guess this question is obsolete any way ;)